charts/kommander-operator/templates/pre_upgrade_kommandercore_hook.yaml

```diff
@@ -0,0 +1,87 @@
+---
+# TODO: remove this job in 2.8
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Chart.Name }}-pre-upgrade
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": before-hook-creation
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ .Chart.Name }}-pre-upgrade
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": before-hook-creation
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
```

Comment:
can we create a `Role` for this instead of user cluster admin? I think it should work since we are only patching an object in the same namespace where this runs

Reply:
I was following the pattern of post_install job. And I think we can also just add `hook-succeeded` to the`hook-delete-policy` to all these RBAC resources? Probably want to do that too for post install job

Reply:
oh actually KommanderCore is a cluster scoped resource so we do need a ClusterRole but ideally we could limit the role to only be able to patch kommandercore

Reply:
Guess gonna address that in a separate PR, wdyt?